### ISSUE TITLE
v0.9.0 Create container fails with mount

### ISSUE THREAD

**Issue by beckend** [blazzy/podman-rest-client#9]
`SpecGenerator` input:

```rust
 dbg!(&item);

  client
    .containers_api()
    .container_create_libpod(item)
    .await
    .map_err(|err| eyre!("Failed to create {name_type} {name_str} - {err:?}"))?;

```

```shell
[packages/podman_cli/src/modules/podman/deploy/handle_containers.rs:88:3] &item = SpecGenerator {
    networks: None,
    annotations: None,
    apparmor_profile: None,
    base_hosts_file: None,
    cap_add: None,
    cap_drop: None,
    cgroup_parent: None,
    cgroupns: None,
    cgroups_mode: None,
    chroot_directories: None,
    cni_networks: None,
    command: None,
    conmon_pid_file: None,
    container_create_command: None,
    create_working_dir: None,
    dependency_containers: None,
    device_cgroup_rule: None,
    devices: None,
    devices_from: None,
    dns_option: None,
    dns_search: None,
    dns_server: None,
    entrypoint: None,
    env: None,
    env_host: None,
    envmerge: None,
    expose: None,
    group_entry: None,
    groups: None,
    health_check_on_failure_action: None,
    healthconfig: None,
    host_device_list: None,
    hostadd: None,
    hostname: None,
    hostusers: None,
    httpproxy: None,
    idmappings: None,
    image: Some(
        "mirror.gcr.io/mccutchen/go-httpbin:latest",
    ),
    image_arch: None,
    image_os: None,
    image_variant: None,
    image_volume_mode: None,
    image_volumes: None,
    init: None,
    init_container_type: None,
    init_path: None,
    intel_rdt: None,
    ipcns: None,
    label_nested: None,
    labels: None,
    log_configuration: None,
    manage_password: None,
    mask: None,
    mounts: Some(
        [
            Mount {
                bind_options: None,
                cluster_options: None,
                consistency: None,
                read_only: None,
                source: None,
                target: None,
                tmpfs_options: None,
                type: None,
                volume_options: None,
            },
        ],
    ),
    name: Some(
        "one",
    ),
    netns: Some(
        Namespace {
            nsmode: Some(
                "host",
            ),
            value: None,
        },
    ),
    network_options: None,
    no_new_privileges: None,
    oci_runtime: None,
    oom_score_adj: None,
    overlay_volumes: None,
    passwd_entry: None,
    personality: None,
    pidns: None,
    pod: None,
    portmappings: None,
    privileged: None,
    procfs_opts: None,
    publish_image_ports: None,
    r_limits: None,
    raw_image_name: None,
    read_only_filesystem: None,
    read_write_tmpfs: None,
    remove: None,
    resource_limits: None,
    restart_policy: Some(
        "unless-stopped",
    ),
    restart_tries: None,
    rootfs: None,
    rootfs_mapping: None,
    rootfs_overlay: None,
    rootfs_propagation: None,
    sdnotify_mode: None,
    seccomp_policy: None,
    seccomp_profile_path: None,
    secret_env: None,
    secrets: None,
    selinux_opts: None,
    shm_size: None,
    shm_size_systemd: None,
    startup_health_config: None,
    stdin: None,
    stop_signal: None,
    stop_timeout: None,
    storage_opts: None,
    sysctl: None,
    systemd: None,
    terminal: None,
    throttle_read_bps_device: None,
    throttle_read_iops_device: None,
    throttle_write_bps_device: None,
    throttle_write_iops_device: None,
    timeout: None,
    timezone: Some(
        "local",
    ),
    umask: None,
    unified: None,
    unmask: None,
    unsetenv: None,
    unsetenvall: None,
    use_image_hosts: None,
    use_image_resolve_conf: None,
    user: Some(
        "65534",
    ),
    userns: None,
    utsns: None,
    volatile: None,
    volumes: None,
    volumes_from: None,
    weight_device: None,
    work_dir: None,
}

Error: Failed to create Container one - Api(ApiError { code: 500, body: "hyper::body::Incoming" })
```

**Comment by beckend** [comment]
It turns out doing this:

```
# TOML config mount.

# source = '/tmp'
# destination = '/tmp_test'
# type = 'bind'
```

Makes podman return 500 response.

**Comment by beckend** [comment]
I see that the generator decided to rename these to whatever Case they liked, so deserialize was troublesome:

https://github.com/blazzy/podman-rest-client/blob/main/podman-autogen-api/src/models/mount.rs#L25-L31

**Comment by beckend** [comment]
```fish
# mount source
❯ ls -ah /home/admin/tmp/
 .   ..   .nobackup   bench   jobs   plot.svg   redb   wireguard   work
```

**Comment by blazzy** [comment]
I haven't had a chance to look at this closely yet, but I suspect you're actually running into this problem from podman: https://github.com/containers/podman/issues/13092

I think solution in the short term might be that we need to hand craft the structs. 😞 

I already massaged one struct [here](https://github.com/blazzy/podman-rest-client/blob/61ca1c16c55ee0224d9753b7042c7643d541b509/podman-autogen-api/src/models/inspect_network_settings.rs#L68) make to make a HashMap key value `Optional`. `golang` really likes its `nil` values.

**Comment by beckend** [comment]
Ok, what shape is it expecting?

**Comment by blazzy** [comment]
This should fix it: https://github.com/blazzy/podman-rest-client/pull/12. Just changing `Target` to `Destination`. There may be other things wrong with the `Mount` structure, but that gets the basics working at least.

> Api(ApiError { code: 500, body: "hyper::body::Incoming" })

These errors were super unhelpful. I made a change in the [generated client](https://github.com/blazzy/podman-rest-client/pull/11) published in v0.10.0 to make it actually parse the error body to this:
 
>Api(ApiError { code: 500, body: Json(Object {"cause": String("container directory cannot be empty"), "message": String("container directory cannot be empty"), "response": Number(500)}) })
